### PR TITLE
ignite 28.2.0 (new formula)

### PR DESCRIPTION
**Pull request by julienrbrt**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/i/ignite.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  url "https://github.com/ignite/cli/archive/refs/tags/v28.2.0.tar.gz"
+  sha256 "556f953fd7f922354dea64e7b3dade5dd75b3f62ece93167e2ba126cac27602e"
+  license "Apache-2.0"
+
+  depends_on "go"
+  depends_on "node"
+
+  def install
+    bin.mkpath
+    system "go", "build", "-mod=readonly", *std_go_args, "-o", bin, "./ignite/cmd/ignite/..."
```
Comment (SMillerDev) [inline Formula/i/ignite.rb:13]: The output can't have the name of the binary directory

File: Formula/i/ignite.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  url "https://github.com/ignite/cli/archive/refs/tags/v28.2.0.tar.gz"
+  sha256 "556f953fd7f922354dea64e7b3dade5dd75b3f62ece93167e2ba126cac27602e"
+  license "Apache-2.0"
+
+  depends_on "go"
+  depends_on "node"
+
+  def install
+    bin.mkpath
+    system "go", "build", "-mod=readonly", *std_go_args, "-o", bin/"ignite", "./ignite/cmd/ignite/..."
```
Comment (SMillerDev) [inline Formula/i/ignite.rb:13]:
```suggestion
    system "go", "build", "-mod=readonly", *std_go_args(output:  bin/"ignite), ".../ignite/cmd/ignite"
```

File: Formula/i/ignite.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  test do
+    # assert scaffolding works
+    ENV["DO_NOT_TRACK"] = "1"
+    system "#{bin}/ignite", "s", "chain", "mars"
```
Comment (SMillerDev) [inline Formula/i/ignite.rb:19]:
```suggestion
    system bin/"ignite", "s", "chain", "mars"
```

File: Formula/i/ignite.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+class Ignite < Formula
+  desc "Build, launch, and maintain any crypto application with Ignite CLI"
+  homepage "https://github.com/ignite/cli"
+  url "https://github.com/ignite/cli/archive/refs/tags/v28.2.0.tar.gz"
+  sha256 "556f953fd7f922354dea64e7b3dade5dd75b3f62ece93167e2ba126cac27602e"
+  license "Apache-2.0"
+
+  depends_on "go"
+  depends_on "node"
```
Comment (SMillerDev) [inline Formula/i/ignite.rb:9]: Are these needed at runtime?

Comment (julienrbrt) [reply Formula/i/ignite.rb:9]: Yes, we need both of them

File: Formula/i/ignite.rb (hunk 3 above)
Comment (p-linnane) [reply Formula/i/ignite.rb]: @SMillerDev This comes up enough that we should probably ask our resident RuboCop champion @issyl0 to take a look 😄

File: Formula/i/ignite.rb (hunk 2 above)
Comment (p-linnane) [reply Formula/i/ignite.rb:13]: @julienrbrt This suggestion needs to be implemented.

File: Formula/i/ignite.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+  def install
+    bin.mkpath
+    system "go", "build", "-mod=readonly", *std_go_args, "-o", bin/"ignite", "./ignite/cmd/ignite/..."
+  end
+
+  test do
+    # assert scaffolding works
```
Comment (p-linnane) [inline Formula/i/ignite.rb:17]:
```suggestion

```

File: Formula/i/ignite.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  homepage "https://github.com/ignite/cli"
+  url "https://github.com/ignite/cli/archive/refs/tags/v28.2.0.tar.gz"
+  sha256 "556f953fd7f922354dea64e7b3dade5dd75b3f62ece93167e2ba126cac27602e"
+  license "Apache-2.0"
+
+  depends_on "go"
+  depends_on "node"
+
+  def install
+    bin.mkpath
```
Comment (p-linnane) [inline Formula/i/ignite.rb:12]:
```suggestion

```

**Review by p-linnane — CHANGES_REQUESTED** [review]
Thanks for your first contribution to Homebrew @julienrbrt. Just a few minor comments to get this to conform with our standard styling.

File: Formula/i/ignite.rb (hunk 2 above)
Comment (p-linnane) [reply Formula/i/ignite.rb:13]: Please do. I'm happy to help debug if it fails.

Comment (julienrbrt) [reply Formula/i/ignite.rb]: Same error, I am getting locally on CI due to the switch from `./ignite/cmd/ignite/...` to `.../ignite/cmd/ignite`.
cc @p-linnane

File: Formula/i/ignite.rb [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  homepage "https://github.com/ignite/cli"
+  url "https://github.com/ignite/cli/archive/refs/tags/v28.2.0.tar.gz"
+  sha256 "556f953fd7f922354dea64e7b3dade5dd75b3f62ece93167e2ba126cac27602e"
+  license "Apache-2.0"
+
+  depends_on "go"
+  depends_on "node"
+
+  def install
+    system "go", "build", "-mod=readonly", *std_go_args(output: bin/"ignite"), "./ignite/cmd/ignite/..."
```
Comment (p-linnane) [inline Formula/i/ignite.rb:12]:
```suggestion
    system "go", "build", "-mod=readonly", *std_go_args(output: bin/"ignite"), "./ignite/cmd/ignite"
```

Comment (p-linnane) [reply Formula/i/ignite.rb:12]: I got this to build fine locally with this change. The test is also successful. If you can push that change we should be good to merge.

**Review by SMillerDev — APPROVED** [review]

**Review by p-linnane — APPROVED** [review]
Thanks!

**Comment by julienrbrt** [comment]
Thank you for all the feedback!

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**